week4_assignment/src/components/Header.tsx

```diff
@@ -0,0 +1,38 @@
+import "./Header.css";
+import {useNavigate} from "react-router-dom";
+import { useState } from "react";
+
+
+export default function Header() {
+    const navigate = useNavigate();
+    
+    const [username, setUsername] = useState("");
+
+    function JoinUsername(event: React.ChangeEvent<HTMLInputElement>) {
+      const inputValue = event.target.value;
+      setUsername(inputValue);
+      localStorage.setItem("username", inputValue); // 로컬 스토리지에 username 저장
+    }  
+
+    function goToPasswordPage() {
+        navigate("/password");  
+    }
+
+    function goToLogin() {
+        navigate("/login");  
+    }
```

Comment:
```suggestion
    const goToPasswordPage = () => {
    navigate("/password");
    };

    const goToLogin = () => {
        navigate("/login");
    };
```

function 키워드를 사용해 함수를 작성해주셨네요!

하지만 가능하면 arrow function을 사용하는 것을 추천드립니다.
Arrow function을 사용하면 다음과 같은 장점이 있습니다:

1.	더 간결한 문법
    함수 선언 방식보다 코드가 짧고 읽기 쉽습니다.

2.	this 바인딩 문제 해결
    Arrow function은 this를 선언 당시의 컨텍스트에 고정하기 때문에, 클래스 컴포넌트에서 발생할 수 있는 this 바인딩 문제를 방지할 수 있습니다.
    
3.	일관성 있는 코드 스타일
    프로젝트 내에서 함수 선언 방식을 통일하면 가독성이 더 좋아집니다.